doc/ot_api_doc.h

```diff
@@ -167,6 +167,7 @@
  *
  * @defgroup plat-alarm               Alarm
  * @defgroup plat-crypto              Crypto - Platform
+ * @defgroup plat-dns                 DNSv6 - Platform
```

Comment:
Would sihhesy just DNS, since there are no "DNSv6" protocol, and DNS is not limited to IPv6 or AAAA.

Reply:
Thanks, I've also changed the label for the non-platform DNS API (it said DNSv6).